QUARKUS-5866.MD

```diff
@@ -0,0 +1,41 @@
+# QUARKUS-5866 - REST client support of TLS configuration updates
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-5866
+
+Upstream PR: https://github.com/quarkusio/quarkus/pull/46961
+
+Our customer reported that hot reload of TLS certificates is not working in OpenShift when Quarkus runs as a native executable.
+From details, we know or assume, they have native applications running in OpenShift and communicating using a REST clients.
+The CA certificate bundle is loaded from mounted secrets and when the served certificate is rotated, the REST client is uses the old one loaded in a memory.
+This is not a bug as we didn't support hot reloads for the Quarkus REST client, and we still don't support it for Quarkus RESTEasy client, Keycloak Admin client and so on.
+While the hot reload will work for programmatically built REST clients if and only if built with the `io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl`, this class is only mentioned once in the documentation in a context of a proxy configuration.
+Therefore, I do not expect that we need to care about programmatically built clients.
+Quarkus implements the clients created via `@RegisterRestClient` using this builder anyway, so no difference.
+
+## Scope of the testing
+Test that Quarkus REST client supports periodic certificate reloading for Quarkus applications run in native and JVM mode.
+We need to run bare-metal tests with the certificates loaded from a local file system, as well as OpenShift serving certificates scenarios.
+Our bare-metal scenarios can play with valid and invalid certificates a bit, to assure that repeated rotation works.
+For the OpenShift scenarios, we will manually rotate the generated service certificate and assure that Quarkus picked up the new certificate and use it.
+Whole reload process is asynchronous and the certificates in the underlying client used by the REST client wrapper are updated after the hot reload has been executed.
+That means that we need to have waited a bit and can't expect everything to work the moment after the CDI event reported successful reload.
+
+## Existing test coverage
+Upstream developers are wont to write JVM-only bare-metal unit tests.
```

Comment:
This line is a bit strange, especially `are wont` part

Reply:
I think that is how it is used, but I'll rewrite it to avoid further discussion.